crates/core/src/query/query_key.rs

```diff
@@ -5,32 +5,46 @@ use std::{
 };
 
 use derive_getters::Getters;
-use derive_more::{Constructor, Display};
+use derive_more::Constructor;
+use enquote::enquote;
 use serde_json::Value;
 
 use super::{apply::InternalError, context::Context, query_arguments::QueryArguments};
 
-pub type OwnedRawKey = RawKey<'static>;
+#[derive(Debug, Clone)]
+pub enum RawKey {
+    Identifier(String),
+    Escaped(String),
+}
 
-#[derive(Debug, Clone, Constructor, Eq, PartialEq, Hash, Display)]
-pub struct RawKey<'a>(pub Cow<'a, str>);
+impl RawKey {
+    pub fn as_str(&self) -> &str {
+        match self {
+            RawKey::Identifier(identifier) => identifier,
+            RawKey::Escaped(escaped) => escaped,
+        }
+    }
+}
 
-impl RawKey<'_> {
-    pub fn into_owned(self) -> OwnedRawKey {
-        RawKey(Cow::Owned(self.0.into_owned()))
+impl Display for RawKey {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            RawKey::Identifier(identifier) => identifier.fmt(f),
+            RawKey::Escaped(escaped) => enquote('\"', escaped).fmt(f),
```

Comment:
maybe enquote does not escape special chars. Check how it gets printed in the dbg! macro, and compare to lsp autocompletion